### ISSUE TITLE
GCE PD actually rounds to GiB

### ISSUE THREAD

**Issue by msau42** [kubernetes/kubernetes#65285]
**Is this a BUG REPORT or FEATURE REQUEST?**:
@kubernetes/sig-storage-bugs 

**What happened**:
Bug https://github.com/kubernetes/kubernetes/issues/56081 thought that PD allocates in GB, perhaps based on the API definition which says GB, but didn't define what GB actually is.  But actually, PD allocates in GiB.

I manually created a PD in GB, and manually attached it to a node:
```
$ gcloud compute disks create msau-test --zone=us-central1-b --size=1GB
```

Ran lsblk on it, and it shows the bytes of the disk are 1GiB:
```
$ lsblk -b
sdc       8:32   0  1073741824  0 disk 
```

**What you expected to happen**:
I think we need to revert https://github.com/kubernetes/kubernetes/pull/56600 (and maybe a few other e2e changes that went in afterwards), and make sure the GCE cloud provider returns disk size in GiB.

**Comment by gnufied** [comment]
In addition to revert and e2e, the resizing code should be fixed to accommodate as well.

**Comment by msau42** [comment]
Resizing logic is [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/gce/gce_disks.go#L751)

**Comment by msau42** [comment]
In the dynamic provisioning e2e tests, there are some lines like this that also need to be reverted: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/volume_provisioning.go#L718

**Comment by msau42** [comment]
Fixed by #66172
